extensions/peer/peer.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	"github.com/hyperledger/fabric/core/committer/txvalidator/plugin"
+	"github.com/hyperledger/fabric/core/committer/txvalidator/v20/plugindispatcher"
+	"github.com/hyperledger/fabric/core/common/sysccprovider"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/protos/common"
+)
+
+type CreateChainFromBlock func(*common.Block, sysccprovider.SystemChaincodeProvider, ledger.DeployedChaincodeInfoProvider,
+	plugindispatcher.LifecycleResources, plugindispatcher.CollectionAndLifecycleResources) error
+
+type CreateChain func(string, ledger.PeerLedger, *common.Block, sysccprovider.SystemChaincodeProvider, plugin.Mapper,
+	ledger.DeployedChaincodeInfoProvider, plugindispatcher.LifecycleResources, plugindispatcher.CollectionAndLifecycleResources) error
+
+//ChannelInitializer is initializer for given channel from persistence
+type ChannelInitializer struct {
+}
+
+func (c *ChannelInitializer) Initialize(cid string, getBlock func(ledger ledger.PeerLedger) (*common.Block, error), createChain CreateChain) error {
+	return nil
+}
+
+//GetChannelInitializer returns initialize channel extension
+func GetChannelInitializer(sysccprovider.SystemChaincodeProvider, plugin.Mapper, ledger.DeployedChaincodeInfoProvider, plugindispatcher.LifecycleResources,
+	plugindispatcher.CollectionAndLifecycleResources) *ChannelInitializer {
+	return &ChannelInitializer{}
+}
+
+//HandleCreateChainFromBlock extends given CreateChainFromBlock handler
+func HandleCreateChainFromBlock(chainID string, handle CreateChainFromBlock) CreateChainFromBlock {
```

Comment:
The name doesn't seem right. This function doesn't actually handle the 'CreateChainFromBlock' - it returns a handler.

Reply:
This function is gone now after refactoring the code.